packages/ra-input-rich-text/src/buttons/ColorButtons.tsx

```diff
@@ -0,0 +1,191 @@
+            }
+        };
+        // Bind the event listener
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => {
+            // Unbind the event listener on clean up
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [ref, onClick]);
+};
+
+type OutsideListenerProps = {
+    className?: string;
+    onClick: () => void;
+    children: React.ReactNode;
+};
+
+/**
+ * Component that listens if you click outside of it
+ */
+const OutsideListener = ({
+    className,
+    onClick,
+    children,
+}: OutsideListenerProps) => {
+    const wrapperRef = React.useRef(null);
+    useOutsideListener(wrapperRef, onClick);
+
+    return (
+        <div className={className} ref={wrapperRef}>
+            {children}
+        </div>
+    );
+};
+
+enum ColorType {
+    FONT = 'font',
+    BACKGROUND = 'background',
+}
+
+interface ColorChoiceDialogProps {
+    editor: any;
+    close: () => void;
+    colorType: ColorType;
+}
+
+const ColorChoiceDialog = ({
+    editor,
+    close,
+    colorType,
+}: ColorChoiceDialogProps) => {
+    const [theme] = useTheme();
+    const colors = [grey, red, orange, yellow, green, blue, purple];
+    const shades = [900, 700, 500, 300, 100];
+
+    const selectColor = (color: string) => {
+        if (colorType === ColorType.FONT) {
+            editor.chain().focus().setColor(color).run();
+        } else {
+            editor.chain().focus().toggleHighlight({ color }).run();
+        }
+        close();
+    };
+
+    return (
+        <Card
+            sx={{
+                position: 'absolute',
+                top: 38,
+                left: colorType === ColorType.FONT ? 0 : '50%',
+                p: 1,
+                border: `1px solid ${theme?.palette?.background?.default}`,
+                display: 'flex',
+                flexDirection: 'column',
+                gap: 1,
+                zIndex: 1,
+            }}
+        >
+            {shades.map((shade, line) => (
+                <Box
+                    key={`shade-${shade}`}
+                    sx={{ display: 'flex', flexDirection: 'row', gap: 1 }}
+                >
+                    {colors.map((color, row) => (
+                        <Box
+                            key={`color-${line * colors.length + row + 1}`}
+                            sx={{
+                                width: 16,
+                                height: 16,
+                                cursor: 'pointer',
+                                // @ts-ignore
+                                backgroundColor: color[shade],
+                            }}
+                            // @ts-ignore
+                            onClick={() => selectColor(color[shade])}
+                        ></Box>
+                    ))}
+                </Box>
+            ))}
+        </Card>
+    );
+};
+
+export const ColorButtons = (props: Omit<ToggleButtonProps, 'value'>) => {
```

Comment:
Would you mind moving this component at the top of the file ? We usually put first the most important things.